test/e2e/resourceinterpreter_test.go

```diff
@@ -422,4 +422,208 @@ end`,
 			}
 		})
 	})
+
+	ginkgo.Context("InterpreterOperation Retain testing", func() {
+		var waitTime time.Duration
+		var updatedPaused bool
+
+		ginkgo.BeforeEach(func() {
+			waitTime = 5 * time.Second
+			updatedPaused = true
+			customization = testhelper.NewResourceInterpreterCustomization(
+				"interpreter-customization"+rand.String(RandomStrLength),
+				configv1alpha1.CustomizationTarget{
+					APIVersion: "apps/v1",
+					Kind:       "Deployment",
+				},
+				configv1alpha1.CustomizationRules{
+					Retention: &configv1alpha1.LocalValueRetention{
+						LuaScript: `
+function Retain(desiredObj, observedObj)
+  desiredObj.spec.paused = observedObj.spec.paused
+  return desiredObj
+end`,
+					},
+				})
+		})
+
+		ginkgo.It("Retain testing", func() {
+			ginkgo.By("update workload's spec.paused to true", func() {
+				clusterClient := framework.GetClusterClient(targetCluster)
+				gomega.Expect(clusterClient).ShouldNot(gomega.BeNil())
+				var memberDeploy *appsv1.Deployment
+				framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+					func(deployment *appsv1.Deployment) bool {
+						memberDeploy = deployment
+						return true
+					})
+				memberDeploy.Spec.Paused = updatedPaused
+				framework.UpdateDeployment(clusterClient, memberDeploy)
+			})
+
+			// Wait executeController to reconcile then check if it is retained
+			time.Sleep(waitTime)
+			ginkgo.By("check if workload's spec.paused is retained", func() {
+				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
+					var memberDeployment *appsv1.Deployment
+					framework.WaitDeploymentPresentOnClusterFitWith(targetCluster, deployment.Namespace, deployment.Name,
+						func(deployment *appsv1.Deployment) bool {
+							memberDeployment = deployment
+							return true
+						})
+					return memberDeployment.Spec.Paused, nil
+				}, pollTimeout, pollInterval).Should(gomega.Equal(updatedPaused))
```

Comment:
It seems strange. You updated the `Pause` field of deployment in the member cluster before, then get it from member cluster. It doesn't prove the script works.

Reply:
At line 465, it sleeps some time, and then checks whether the value on the cluster is overwritten.